src/csync/csync_update.cpp

```diff
@@ -522,6 +525,25 @@ int csync_ftw(CSYNC *ctx, const char *uri, csync_walker_fn fn,
   int rc = 0;
 
   bool do_read_from_db = (ctx->current == REMOTE_REPLICA && ctx->remote.read_from_db);
+  const char *db_uri = uri;
+
+  if (ctx->current == LOCAL_REPLICA
+      && ctx->local_discovery_style == LocalDiscoveryStyle::DatabaseAndFilesystem) {
+      const char *local_uri = uri + strlen(ctx->local.uri);
+      if (*local_uri == '/')
+          ++local_uri;
+      db_uri = local_uri;
+      do_read_from_db = true;
+
+      // Minor bug: local_uri doesn't have a trailing /. Example: Assume it's "d/foo"
+      // and we want to check whether we should read from the db. Assume "d/foo a" is
+      // in locally_touched_dirs. Then this check will say no, don't read from the db!
+      // (because "d/foo" < "d/foo a" < "d/foo/bar")
+      auto it = ctx->locally_touched_dirs.lower_bound(QByteArray(local_uri));
```

Comment:
The QByteArray allocation could be avoided in C++14, there is an overload of lower_bound that can work for types for which the comparator is transparent.

But that would imply a change to the makefile to enable C++14.
Maybe it is not needed as it should not happen so often anyway.

Reply:
I'll add a note about this for C++14, but won't enable it in this PR.